### PR TITLE
[DXCDT-110] [DXCDT-113] Refactor grant, guardian tests

### PR DESCRIPTION
**Pull request by sergiughf**
## Description

Refactor grant, guardian tests so they can be ran in any order and they clean up the test tenant after running.

## Testing

- [x] This change adds test coverage for new/changed/fixed functionality

## Checklist

- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I have reviewed my own code beforehand.
- [x] I have added documentation for new/changed functionality in this PR.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [x] The correct base branch is being used, if not `main`.

### CODE REVIEW

**Review by willvedd — APPROVED** [review]
Great work.

File: management/user_test.go [hunk 1]
```diff
@@ -465,3 +466,22 @@ func TestUserIdentity(t *testing.T) {
 		}
 	})
 }
+
+func givenAUser(t *testing.T) *User {
+	user := &User{
+		Connection: auth0.String("Username-Password-Authentication"),
+		Email:      auth0.String("chuck@chucknorris.com"),
```
Comment (Widcket) [inline management/user_test.go:473]:
```suggestion
		Email:      auth0.String("chuck@example.com"),
```
Replace the domain name with a IANA-reserved domain name (https://www.rfc-editor.org/rfc/rfc6761.html#section-6.5) as chucknorris.com is an actual domain.

File: management/guardian_test.go [hunk 2]
```diff
@@ -12,267 +13,314 @@ import (
 func TestGuardian(t *testing.T) {
 	t.Run("MultiFactor", func(t *testing.T) {
 		t.Run("List", func(t *testing.T) {
+			const expectedGuardianFactors = 8
```
Comment (Widcket) [inline management/guardian_test.go:16]: Where does this number come from? If it's from the API, shouldn't we assert that it is > 0 so the tests don't break when new factors are added, or some get dropped?

Comment (sergiughf) [reply management/guardian_test.go]: It's just from the API:) good point! I updated this in [52df276](https://github.com/auth0/go-auth0/pull/53/commits/52df276c920f66ce5fa72a33cd9dc367fe3929bf)

File: management/guardian_test.go [hunk 3]
```diff
@@ -13,266 +14,312 @@ func TestGuardian(t *testing.T) {
...
 
 			t.Run("Enable", func(t *testing.T) {
-				err := m.Guardian.MultiFactor.Phone.Enable(false)
-				if err != nil {
-					t.Error(err)
-				}
+				initialStatus, err := getInitialMFAStatus("sms")
+				assert.NoError(t, err)
+
+				t.Cleanup(func() {
```
Comment (Widcket) [inline management/guardian_test.go:61]: Just curious, why use a cleanup method here and not in the other cases?

Comment (Widcket) [reply management/guardian_test.go:61]: Oh, I see some other test cases use the cleanup method, but there are ones that aren't. Is this intended?

Comment (sergiughf) [reply management/guardian_test.go:65]: Good catch, they should all consistently use the cleanup method at the beginning of the func. Fixed in [69f7d42](https://github.com/auth0/go-auth0/pull/53/commits/69f7d4270c8b1942f8ec21cc56f50ec518bdb5e6)

Also should have probably attached the cleanup on all the givens on previous tests as well to simplify the test logic. Will do a final sweep once all resources tests are updated to clean this up.

Comment (sergiughf) [reply management/guardian_test.go:65]: e.g. https://github.com/auth0/go-auth0/pull/53/files#diff-eeb45118a10f1f40839fd8728d64f3ba285a252bc385d9e73c7159529fe68110R481-R483

**Review by Widcket — APPROVED** [review]
LGTM

File: management/guardian_test.go (hunk 3 above)
Comment (Widcket) [reply management/guardian_test.go:65]: Yeah, good idea.

**Pull request merged**